jupyterlab_git/git.py

```diff
@@ -485,13 +486,22 @@ async def log(self, path, history_count=10, follow_path=None):
             return {"code": code, "command": " ".join(cmd), "message": my_error}
 
         result = []
-        if is_single_file:
-            # an extra newline get outputted when --numstat is used
-            my_output = my_output.replace("\n\n", "\n")
         line_array = my_output.splitlines()
-        i = 0
+
+        if is_single_file:
+            parsed_lines = []
+            for line in line_array:
+                if "\0" in line:
+                    split_line = line.replace("\0", "\t").split("\t")
+                    parsed_lines.append("\t".join(split_line[:-1]))
+                    if split_line[-1]:
+                        parsed_lines.append(split_line[-1])
+                else:
+                    parsed_lines.append(line)
+            line_array = parsed_lines
```

Comment:
Initially I just wanted to use strip_and_split with every separator being `\0`, but this was causing trouble since the number of lines is 4 if its a regular single file commit, but 5 if its a move/relocation one. Doing it this way the line_array is in the form

```python
[
  "Commit",
  "Author",
  "Date"
  "Insertions\tDeletions\tOld_file_path\tNew_file_path",
  "Pre_commit",
  ...
]
```
so now at least the `PREVIOUS_COMMIT_OFFSET` will be constant, like it was before.

Hoping to make this block smaller through list comprehension, but atm I think its fine.